packages/data-provider/src/utils/buildQuery.ts

```diff
@@ -0,0 +1,123 @@
+import { SQL, and, not, or, sql } from 'drizzle-orm';
+import * as _ from 'lodash-es';
+import { BadRequest } from './errors.js';
+import {
+	ArrayFilterValue,
+	CombinationKeys,
+	CombinationOperator,
+	FilterOperator,
+	GreaterThanFilter,
+	LesserThanFilter,
+	Operator,
+	isArrayFilter,
+	isCombination,
+	isFilter,
+} from '@overture-stack/sqon-builder';
+
+const jsonbColumnName = 'data';
+
+export const isGreaterThanFilter = (operator: Operator): operator is GreaterThanFilter =>
+	GreaterThanFilter.safeParse(operator).success;
+
+export const isLesserThanFilter = (operator: Operator): operator is LesserThanFilter =>
+	LesserThanFilter.safeParse(operator).success;
+
+// Map the array and format each element based on its type
+const formatForSQL = (value: ArrayFilterValue) => {
+	if (Array.isArray(value)) {
+		// Handle array of strings or numbers
+		return value
+			.map((element) => {
+				if (typeof element === 'string') {
+					return `'${element}'`; // Surround strings with single quotes
+				} else if (typeof element === 'number') {
+					return element.toString(); // Numbers don't need quotes
+				} else {
+					throw new BadRequest(`Invalid SQON format. Unsupported data type: ${typeof element}`);
+				}
+			})
+			.join(', ');
+	} else if (typeof value === 'string') {
+		// Handle single string
+		return value;
+	} else if (typeof value === 'number') {
+		// Handle single number
+		return value;
+	} else {
+		throw new BadRequest(`Invalid SQON. Unsupported data type: ${typeof value}`);
+	}
+};
+
+const processFilterOperator = (operator: FilterOperator): SQL<unknown> => {
+	const { fieldName, value } = operator.content;
+	if (isArrayFilter(operator)) {
+		// op is in
+		return sql.raw(`${jsonbColumnName} ->> '${formatForSQL(fieldName)}' IN (${formatForSQL(value)})`);
+	}
+
+	// is an scalar filter op is lt or gt
+	if (isGreaterThanFilter(operator)) {
+		return sql.raw(`${jsonbColumnName} ->> '${formatForSQL(fieldName)}' > '${formatForSQL(value)}'`);
+	} else if (isLesserThanFilter(operator)) {
+		return sql.raw(`${jsonbColumnName} ->> '${formatForSQL(fieldName)}' < '${formatForSQL(value)}'`);
+	}
+
+	throw new BadRequest(`Invalid SQON format. Unsupported SQON filter operator`);
+};
+
+const iterateOperator = (operator: Operator): SQL<unknown> => {
+	if (isFilter(operator)) {
+		// op in [in, lt, gt]
+		return processFilterOperator(operator);
+	} else if (isCombination(operator)) {
+		// op in [and, or, not]
+		return processCombinationOperator(operator);
+	}
+	throw new BadRequest(`Invalid SQON format. Unsupported SQON Operator`);
```

Comment:
consistent empty spacing 👍 
```suggestion

	throw new BadRequest(`Invalid SQON format. Unsupported SQON Operator`);
```